test/e2e/Apps/BasicDotNetIsolated/TerminateOrchestration.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Net;
+using Grpc.Core;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.DurableTask;
+using Microsoft.DurableTask.Client;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Durable.Tests.E2E
+{
+    public static class LongRunningOrchestration
+    {
+        [Function(nameof(LongRunningOrchestrator))]
+        public static async Task<List<string>> LongRunningOrchestrator(
+            [OrchestrationTrigger] TaskOrchestrationContext context)
+        {
+            ILogger logger = context.CreateReplaySafeLogger(nameof(HelloCities));
+            logger.LogInformation("Starting long-running orchestration.");
+            var outputs = new List<string>();
+
+            // Call our fake activity 100,000 times to simulate an orchestration that might run for >= 10,000s (2.7 hours)
+            for (int i = 0; i < 100000; i++) 
```

Comment:
I worry that a large fan-out like this may make the test unstable on the low-powered CI machines due to the memory and I/O requirements. Do we need a fan-out at all or could we just use an orchestration that sleeps for a long time or waits for an external event?

Reply:
This isn't a fan-out though, the orchestrations are called sequentially. I agree, it's overkill though since most of the tests run in well under a minute, I can shorten it, the idea behind writing the orchestrator this way was to demonstrate what might cause an orchestration with problematic runtime.

Reply:
Oh, sorry, I misread the code. OK, I'm less worried about this then.